### ISSUE TITLE
[HifiveUnleashed] Fatal error in ../../src/codegen/safepoint-table.cc, line 62

### ISSUE THREAD

**Issue by lazyparser** [v8-riscv/v8#129]
**Describe the bug**
The Good News is that currently I see one error occurred 3 times in Octane benchmark.

**To Reproduce**
1. Build with options https://github.com/isrc-cas/PLCT-Toolbox/blob/master/init_v8_all_in_one.sh
2. Run test/program `cd test/benchmarks/csuite && ./csuite.py -r 3 octane baseline ~/riscv64.native.debug/d8`
3. See error

**Expected behavior**
Happy Ending with a perf report.

**Errors/Logs**
If applicable, copy/paste the errors or log messages here.

**Additional context**
```
[wuwei@fedora-riscv-sifive csuite]$ ./csuite.py -r 3 octane baseline ~/riscv64.native.debug/d8
Normally, octane requires 10 runs to get stable results.


#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x3fff93c1a8
==== C stack trace ===============================

    /home/wuwei/riscv64.native.debug/d8(+0x1b2512c) [0x2aaf79912c]
    /home/wuwei/riscv64.native.debug/d8(+0x19215aa) [0x2aaf5955aa]
    /home/wuwei/riscv64.native.debug/d8(+0x1917dbc) [0x2aaf58bdbc]
    /home/wuwei/riscv64.native.debug/d8(+0x2d4d1a) [0x2aadf48d1a]
    /home/wuwei/riscv64.native.debug/d8(+0x5c12bc) [0x2aae2352bc]
    /home/wuwei/riscv64.native.debug/d8(+0x38bdcc) [0x2aadfffdcc]
    /home/wuwei/riscv64.native.debug/d8(+0x394c6e) [0x2aae008c6e]
    /home/wuwei/riscv64.native.debug/d8(+0x41d8b0) [0x2aae0918b0]
    /home/wuwei/riscv64.native.debug/d8(+0x41da66) [0x2aae091a66]
    /home/wuwei/riscv64.native.debug/d8(+0x4afc7a) [0x2aae123c7a]
    /home/wuwei/riscv64.native.debug/d8(+0x4199ae) [0x2aae08d9ae]
    /home/wuwei/riscv64.native.debug/d8(+0x43fee6) [0x2aae0b3ee6]
    /home/wuwei/riscv64.native.debug/d8(+0x4405a4) [0x2aae0b45a4]
    /home/wuwei/riscv64.native.debug/d8(+0x44304e) [0x2aae0b704e]
    /home/wuwei/riscv64.native.debug/d8(+0x443198) [0x2aae0b7198]
    /home/wuwei/riscv64.native.debug/d8(+0x3fbe0a) [0x2aae06fe0a]
    /home/wuwei/riscv64.native.debug/d8(+0x8ac6b0) [0x2aae5206b0]
    /home/wuwei/riscv64.native.debug/d8(+0x8b723c) [0x2aae52b23c]
    /home/wuwei/riscv64.native.debug/d8(+0x126246c) [0x2aaeed646c]
/bin/sh: line 1: 91437 Trace/breakpoint trap   (core dumped) /home/wuwei/riscv64.native.debug/d8 --expose-gc run.js > /home/wuwei/test/benchmarks/csuite/_benchmark_runner_data/out.0.txt


#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x3fff8f45a8
==== C stack trace ===============================

    /home/wuwei/riscv64.native.debug/d8(+0x1b2512c) [0x2abae9812c]
    /home/wuwei/riscv64.native.debug/d8(+0x19215aa) [0x2abac945aa]
    /home/wuwei/riscv64.native.debug/d8(+0x1917dbc) [0x2abac8adbc]
    /home/wuwei/riscv64.native.debug/d8(+0x2d4d1a) [0x2ab9647d1a]
    /home/wuwei/riscv64.native.debug/d8(+0x5c12bc) [0x2ab99342bc]
    /home/wuwei/riscv64.native.debug/d8(+0x38bdcc) [0x2ab96fedcc]
    /home/wuwei/riscv64.native.debug/d8(+0x394c6e) [0x2ab9707c6e]
    /home/wuwei/riscv64.native.debug/d8(+0x41d8b0) [0x2ab97908b0]
    /home/wuwei/riscv64.native.debug/d8(+0x41da66) [0x2ab9790a66]
    /home/wuwei/riscv64.native.debug/d8(+0x4afc7a) [0x2ab9822c7a]
    /home/wuwei/riscv64.native.debug/d8(+0x4199ae) [0x2ab978c9ae]
    /home/wuwei/riscv64.native.debug/d8(+0x43fee6) [0x2ab97b2ee6]
    /home/wuwei/riscv64.native.debug/d8(+0x4405a4) [0x2ab97b35a4]
    /home/wuwei/riscv64.native.debug/d8(+0x44304e) [0x2ab97b604e]
    /home/wuwei/riscv64.native.debug/d8(+0x443198) [0x2ab97b6198]
    /home/wuwei/riscv64.native.debug/d8(+0x3fbe0a) [0x2ab976ee0a]
    /home/wuwei/riscv64.native.debug/d8(+0x8ac6b0) [0x2ab9c1f6b0]
    /home/wuwei/riscv64.native.debug/d8(+0x8b723c) [0x2ab9c2a23c]
    /home/wuwei/riscv64.native.debug/d8(+0x126246c) [0x2aba5d546c]
/bin/sh: line 1: 91996 Trace/breakpoint trap   (core dumped) /home/wuwei/riscv64.native.debug/d8 --expose-gc run.js > /home/wuwei/test/benchmarks/csuite/_benchmark_runner_data/out.1.txt


#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x3fffe1f3d8
==== C stack trace ===============================

    /home/wuwei/riscv64.native.debug/d8(+0x1b2512c) [0x2ac72ff12c]
    /home/wuwei/riscv64.native.debug/d8(+0x19215aa) [0x2ac70fb5aa]
    /home/wuwei/riscv64.native.debug/d8(+0x1917dbc) [0x2ac70f1dbc]
    /home/wuwei/riscv64.native.debug/d8(+0x2d4d1a) [0x2ac5aaed1a]
    /home/wuwei/riscv64.native.debug/d8(+0x5c12bc) [0x2ac5d9b2bc]
    /home/wuwei/riscv64.native.debug/d8(+0x38bdcc) [0x2ac5b65dcc]
    /home/wuwei/riscv64.native.debug/d8(+0x394c6e) [0x2ac5b6ec6e]
    /home/wuwei/riscv64.native.debug/d8(+0x41d8b0) [0x2ac5bf78b0]
    /home/wuwei/riscv64.native.debug/d8(+0x41da66) [0x2ac5bf7a66]
    /home/wuwei/riscv64.native.debug/d8(+0x4afc7a) [0x2ac5c89c7a]
    /home/wuwei/riscv64.native.debug/d8(+0x4199ae) [0x2ac5bf39ae]
    /home/wuwei/riscv64.native.debug/d8(+0x43fee6) [0x2ac5c19ee6]
    /home/wuwei/riscv64.native.debug/d8(+0x4405a4) [0x2ac5c1a5a4]
    /home/wuwei/riscv64.native.debug/d8(+0x44304e) [0x2ac5c1d04e]
    /home/wuwei/riscv64.native.debug/d8(+0x443198) [0x2ac5c1d198]
    /home/wuwei/riscv64.native.debug/d8(+0x3fbe0a) [0x2ac5bd5e0a]
    /home/wuwei/riscv64.native.debug/d8(+0x8ac6b0) [0x2ac60866b0]
    /home/wuwei/riscv64.native.debug/d8(+0x8b723c) [0x2ac609123c]
    /home/wuwei/riscv64.native.debug/d8(+0x126246c) [0x2ac6a3c46c]
/bin/sh: line 1: 92435 Trace/breakpoint trap   (core dumped) /home/wuwei/riscv64.native.debug/d8 --expose-gc run.js > /home/wuwei/test/benchmarks/csuite/_benchmark_runner_data/out.2.txt
Traceback (most recent call last):
  File "/home/wuwei/test/benchmarks/csuite/benchmark.py", line 223, in <module>
    br.Run()
  File "/home/wuwei/test/benchmarks/csuite/benchmark.py", line 139, in Run
    self.ProcessOutput(suite)
  File "/home/wuwei/test/benchmarks/csuite/benchmark.py", line 184, in ProcessOutput
    suite.ComputeScore()
  File "/home/wuwei/test/benchmarks/csuite/benchmark.py", line 100, in ComputeScore
    self.ComputeScoreV8Octane(self.name)
  File "/home/wuwei/test/benchmarks/csuite/benchmark.py", line 89, in ComputeScoreV8Octane
    self.score = self.avgresult[found_name]
KeyError: ''
Wrote /home/wuwei/test/benchmarks/csuite/_results/master.
Run octane again with compare mode to see results.
[wuwei@fedora-riscv-sifive csuite]$
```
the files mentioned like
`/home/wuwei/test/benchmarks/csuite/_benchmark_runner_data/out.0.txt`
is just time score:

```
Richards: xxx
DeltaBlue: xxx
Crypto: xxx
RayTrace: xxx
```

**Comment by lazyparser** [comment]
I think it is a known bug. Is @LOGQTAINIA working on the issue?

```
$ git log -1
commit b8c4668d908f9ae05edd5bc4f0cab25bd4e13931 (HEAD -> riscv-porting-dev, riscv/riscv-porting-dev)
Author: luyahan <yahan@iscas.ac.cn>
Date:   Tue Aug 4 16:12:54 2020 +0800

    Using F Standard Extension instruct proces float32 value in AssembleMove.
        Resolves #45.
        On riscv64, float32 value will be NaN boxed by fmov.s.x/flw. So we
    should using it when move a float32 value.
```

**Comment by lazyparser** [comment]
Ohh, I got it wrong:

the octane benchmark quits when first crash hit. and I set to run 3 times for average run time.

**Comment by lazyparser** [comment]
seems like that it fails on sim build too.

```
wuwei@plct-build-5:~/v8riscv/v8$ cd test/benchmarks/csuite && ./csuite.py -r 1 octane baseline ~/v8riscv/v8/out/riscv64.sim/d8
Normally, octane requires 10 runs to get stable results.


#
# Fatal error in ../../src/codegen/safepoint-table.cc, line 62
# unreachable code
#
#
#
#FailureMessage Object: 0x7fffadcfd9c0
==== C stack trace ===============================

    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x1d94273) [0x564a0f5b9273]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x1d932bd) [0x564a0f5b82bd]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x1d8a4f3) [0x564a0f5af4f3]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x65f512) [0x564a0de84512]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x97a83e) [0x564a0e19f83e]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x70b55c) [0x564a0df3055c]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x717379) [0x564a0df3c379]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7b68a1) [0x564a0dfdb8a1]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7b5e59) [0x564a0dfdae59]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x8433be) [0x564a0e0683be]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7ad298) [0x564a0dfd2298]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7a9c55) [0x564a0dfcec55]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7a77b5) [0x564a0dfcc7b5]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7b8624) [0x564a0dfdd624]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7b87f5) [0x564a0dfdd7f5]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x768b1a) [0x564a0df8db1a]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xcbe87a) [0x564a0e4e387a]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xf814bc) [0x564a0e7a64bc]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xf80280) [0x564a0e7a5280]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xf8fe7b) [0x564a0e7b4e7b]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xf8ff8b) [0x564a0e7b4f8b]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0xf90534) [0x564a0e7b5534]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x706556) [0x564a0df2b556]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x7050c3) [0x564a0df2a0c3]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x4d9d56) [0x564a0dcfed56]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x4a3854) [0x564a0dcc8854]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x4af3c7) [0x564a0dcd43c7]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x4b1ab1) [0x564a0dcd6ab1]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(+0x4b3c3b) [0x564a0dcd8c3b]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f5ae07eab97]
    /home/wuwei/v8riscv/v8/out/riscv64.sim/d8(_start+0x2a) [0x564a0dcc190a]
Received signal 4 ILL_ILLOPN 564a0f5b60c2
Illegal instruction
Traceback (most recent call last):
  File "/home/wuwei/v8riscv/v8/test/benchmarks/csuite/benchmark.py", line 223, in <module>
    br.Run()
  File "/home/wuwei/v8riscv/v8/test/benchmarks/csuite/benchmark.py", line 139, in Run
    self.ProcessOutput(suite)
  File "/home/wuwei/v8riscv/v8/test/benchmarks/csuite/benchmark.py", line 184, in ProcessOutput
    suite.ComputeScore()
  File "/home/wuwei/v8riscv/v8/test/benchmarks/csuite/benchmark.py", line 100, in ComputeScore
    self.ComputeScoreV8Octane(self.name)
  File "/home/wuwei/v8riscv/v8/test/benchmarks/csuite/benchmark.py", line 89, in ComputeScoreV8Octane
    self.score = self.avgresult[found_name]
KeyError: ''
Wrote /home/wuwei/v8riscv/v8/test/benchmarks/csuite/_results/master.
Run octane again with compare mode to see results.
```

**Comment by penguinwu** [comment]
Look like a duplicate of #56. May be easier to debug #56 first.

**Comment by luyahan** [comment]
I fix it by #139
